### ISSUE TITLE
Question: App navigation using app-route and neon-animated-pages

### ISSUE THREAD

**Issue by jenishngl** [PolymerElements/app-route#134]
### Description

Problem while navigating within app

I am using `app-route` along with `neon-animated-pages` for app navigation
Now I have a link like `http://domain/` and then I navigate to 
`http://domain/view1` and then to `http://domain/view1/user`
### Expected outcome

Now when I click on back button in the browser, it should take me to `http://domain/view1`
### Actual outcome

But it takes way back to `http://domain/` skipping `http://domain/view1` inbetween.
### When it Occurs

When I navigate from `http://domain/` --> `http://domain/view1` --> `http://domain/view1/user` with 1 sec interval in each route
### When it Does not occurs

When I navigate from `http://domain/` --> `http://domain/view1` --> `http://domain/view1/user` with more than 2 sec interval in each route
### Questions

Is it a browser problem or what?
Is there any certain time interval before the browser can actually register the route changes?
Can someone explain me what is happening?

Even used `window.history.back()` to navigate back, but still see the same problem
### Browsers Affected

- [x] Chrome
- [x] Firefox
- [x] Safari 9 //Havent Checked
- [ ] Safari 8 //Havent Checked
- [ ] Safari 7 //Havent Checked
- [x] Edge
- [x] IE 11
- [ ] IE 10 //Havent Checked

**Comment by inyigo** [comment]
Also happening in Safari 9

**Comment by rictic** [comment]
This is controlled by iron-location's dwellTime property: https://elements.polymer-project.org/elements/iron-location

The logic is that *typically* any state that the user wasn't on long enough to mentally register shouldn't be a part of their history. This is intended to prevent accidental history flooding, which can make history useless for the user.

I'm just now noticing that we don't actually expose the `iron-location`'s dwellTime property in `app-location` though! That's a bug and should be fixed with #135.

Until that lands you can work around it using the same underlying apis that `app-location` and `iron-location` use. The difference is between `window.history.pushState` and `window.history.replaceState`, when the url is updated with the former you will get a new history entry, with the latter you won't. The polymer routing elements use the `location-changed` event on `window` to communicate url changes. So this will navigate to `/view1` on the current domain, add an entry into the history, and ensure that the routing elements are aware of the new URL:

```javascript
window.history.pushState({}, null, '/view1');
window.dispatchEvent(new CustomEvent('location-changed'));
```
